### PR TITLE
Astropy Roadmap

### PR DESCRIPTION
**Pull request by ceb8**
This PR adds the Astropy Roadmap document written by the Astropy Roadmap Working Group.    

This PR will be open for a two week comment period, **November 4rd-18th**. The Astropy Roadmap document is the result of a large collaborative effort, and is designed to capture important top level Astropy goals.

Running notes for the Roadmap working group can be found here: https://docs.google.com/document/d/1oKOCJjhsAmLXJyGgkeK85dZYCj7r0v3rJEJhil9QlRg/edit?usp=sharing

### CODE REVIEW

**Comment by jdswinbank** [comment]
Thanks to everybody for their hard work on this!

I think it would be really helpful to provide a bit more context on this roadmap and what it's intended to capture. I'm pretty sure it was extensively discussed at the Coordination Meeting earlier this year, but folks who weren't involved in that meeting might not really understand what it is that we're looking at. Perhaps there have been earlier announcements/public discussion on mailing lists; if somebody could provide a few links for orientation, that would  be great.

My immediate questions on looking at the document are:

- It refers to the “Roadmap 2021 Project Board”, but most of the items in the document don't appear on that board. Is that expected?
- If this is — as that board implies — a roadmap for 2021, should we be worried that it is now November and the vast majority of items still have status “orange” or “red”?

(I suspect what's happening is that this is a roadmap into the far future, and only a few items are identified for action in 2021... but a bit more context could help me figure that out for sure! :-))

**Comment by ceb8** [comment]
@jdswinbank  Good questions. Here are some answers:

- The Roadmap is meant to capture high level actionable items that we as a project aim to undertake to improve the health and stability of the Astropy project. 
- The Roadmap project board will get populated after the roadmap is accepted, once accepted the Roadmap will be a static document, while the project board will be a living document that links the roadmap to specific issues and efforts.  Once this PR is passed if you have an issue that is relevant, put it on the project board. Ongoing discussion about the project board will happen in the \#project channel of Astropy slack.
- This is the 2021 roadmap in that it was written in 2021 not that it catalogs tasks we plan to complete in 2021. The idea is to revisit this roadmap regularly at the Astropy coordination meetings to keep track of progress and write new versions as needed. The hope is that items move from red to orange to green and then off the roadmap, not that everything on it is completely finished before the next update.

**Comment by jdswinbank** [comment]
Thanks; that's really useful!

File: roadmap/roadmap.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+
+## Status Legend
+
+### Green :green_circle:
+- Group or person leading effort is identified and sufficiently supported.
+- Effort is already underway and/or expected to be implemented in the “near term.”
+
+### Orange :large_orange_diamond:    
+- Item is already defined (e.g., issue, APE exists), but not sufficiently underway.
```
Comment (jdswinbank) [inline roadmap/roadmap.md:19]: How is “sufficiently” defined? For example, if somebody has started working on an item, but somebody else thinks the progress they are making is too slow, is it “sufficiently underway”?

File: roadmap/roadmap.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+- Needs more people and/or resources to make it happen. 
+  - *For example, funding applied for, but not awarded.*
+  
+### Red :red_square:  
+- Consensus of the community is that the item is a priority.
+- Item won’t happen without more people/resources.
```
Comment (jdswinbank) [inline roadmap/roadmap.md:25]: How is this different from “needs more people and/or resources to make it happen” (above)?

Comment (jdswinbank) [reply roadmap/roadmap.md:25]: The examples sort of make the distinction more clear, but I suggest rewording so the examples aren't necessary. Maybe something like:

(Orange): Attempts to secure the necessary resources are underway.
(Red): Attempts to secure the necessary resources have not yet been started.

File: roadmap/roadmap.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  - *For example, there’s no one available to write a funding proposal or lead recruitment efforts.*
+  
+## Roadmap 2021 Project Board
+
+Issues relevant to the Roadmap goals can be added and organized on the [Astropy Roadmap 2021 Project Board](https://github.com/orgs/astropy/projects/7). The goal of the project board is to give a current overview of the status of our progress and provide access to the details of the work associated with roadmap items. It will be updated continuously.
+
+## Functionality
+
+- :large_orange_diamond: Provide next-generation spectroscopic reduction, analysis, and visualization tools usable by individual researchers and larger surveys.
```
Comment (jdswinbank) [inline roadmap/roadmap.md:34]: As a non-specialist in the field, “next-generation” seems a bit fuzzy. How do next-generation tools differ from current-generation tools?

File: roadmap/roadmap.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+- :large_orange_diamond: Improve support and validation for input/output of standard file formats used in Astronomy including FITS, ASDF, HDF5, and ECSV as well as compression handling and enhanced support for large tables.
+
+- :large_orange_diamond: Improve interoperability and compatibility, while reducing duplication, amongst Astropy-affiliated packages and other community standard software.
+
+## Hardware and Performance
+
+- :large_orange_diamond: Improve Astropy ecosystem stability in High-Performance Computing environments, for example, and expand options for compatibility with read-only filesystems.
```
Comment (jdswinbank) [inline roadmap/roadmap.md:42]: I'm not sure what the “for example” means here.

**Comment by jdswinbank** [comment]
Overall, this seems like a really useful document! I made a few line-by-line comments, but here are a couple of general thoughts:

- Many of the objectives are quite vaguely specified. For example, almost anything can be counted as a success when the goal is as non-specific “improve X”. That's likely deliberate — even modest improvements are welcome! But given that, this doesn't really read like a roadmap (which would have concrete milestones, etc), so much as a list of important focus areas.
- @ceb8 's comment above was very helpful in understanding the context here. I suggest we should build on that by including information either in this `roadmap.md` file, or in some other accompanying document, about the process surrounding this roadmap — how will it be used? How does it relate to the issue tracker? (How) will it be reviewed and updated? etc.

File: roadmap/roadmap.md (hunk 4 above)
Comment (ceb8) [reply roadmap/roadmap.md:42]: I think it's a typo, leftover text from a previous draft. I've removed it.

File: roadmap/roadmap.md (hunk 1 above)
Comment (pllim) [reply roadmap/roadmap.md:19]: This should probably be left vague and determined on a case-by-case basis. Stale timer might be different for a large complex item compared to a simpler one... 💭

File: roadmap/roadmap.md (hunk 3 above)
Comment (pllim) [reply roadmap/roadmap.md:34]: Personal opinion: Current generation is more like PyRAF scripts, home-grown code that is very specialized to a particular researcher; while next-generation is like Star Trek utopia that allows interoperability and reproducible science.

**Comment by hamogu** [comment]
Not having been part of the road-map working group I really want to comment some of @jdswinbank 's comments: For me to understand this document, it really needs a short intro which can essentially be copied and pasted from https://github.com/astropy/astropy-project/pull/220#issuecomment-961048078

**Review by hamogu — APPROVED** [review]
